### PR TITLE
feat: download sourcery via SPM

### PR DESCRIPTION
**Pull request by caldrian**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Until now there are Sourcery build phases which depend on Sourcery being installed globally. The version of Sourcery was not pinned.

### Solutions

A specific version of Sourcery will be downloaded via SPM.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -41,10 +41,6 @@ excluded:
     - "wire-ios-protos/Protos/messages.pb.swift"
     - "wire-ios-protos/Protos/mls.pb.swift"
     - "wire-ios-protos/Protos/otr.pb.swift"
-    - "wire-ios-data-model/sourcery/generated/AutoMockable.generated.swift"
-    - "wire-ios-sync-engine/sourcery/generated/AutoMockable.generated.swift"
-    - "wire-ios-sync-engine/sourcery/generated/AutoMockable.manual.swift"
-    - "wire-ios-request-strategy/sourcery/generated/AutoMockable.generated.swift"
```
Comment (markusfassbender) [inline .swiftlint.yml]: What is the idea to run SwiftLint on generated files? Anyway they are never programmed by a developer or do I miss something?

File: wire-ios-request-strategy/WireRequestStrategy.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1773,10 +1773,10 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 1669017E1D707509000FE4AF /* Build configuration list for PBXNativeTarget "WireRequestStrategy" */;
 			buildPhases = (
+				630D41B52B07851000633867 /* Run Sourcery */,
 				166901651D707509000FE4AF /* Sources */,
 				166901661D707509000FE4AF /* Frameworks */,
 				166901671D707509000FE4AF /* Headers */,
-				630D41B52B07851000633867 /* Run Sourcery */,
 				668E958827425E0A0028C1F8 /* Run Swiftlint */,
```
Comment (markusfassbender) [inline wire-ios-request-strategy/WireRequestStrategy.xcodeproj/project.pbxproj:1774]: 🤘

**Review by markusfassbender — APPROVED** [review]

File: .swiftlint.yml (hunk 1 above)
Comment (caldrian) [reply .swiftlint.yml]: The files are committed to the repository after being generated, so they are there like any other manually created file. That's why I had thought they could as well just conform to our code style.

However, I decided to revert it: 5a8d9aa5c0fbcd8ab0ffc0b10950c128b065d6f0

Comment (markusfassbender) [reply .swiftlint.yml]: Okay :)

**Review by netbe — APPROVED** [review]
Thanks @caldrian !

**Comment by caldrian** [comment]
@netbe  I wonder why the test results show 1541 removed tests.

**Pull request merged**